chrome/content/zotero/xpcom/data/item.js

```diff
@@ -4398,7 +4398,9 @@ Zotero.Item.prototype.getColoredTags = function () {
 	let tagColors = Zotero.Tags.getColors(this.libraryID);
 	for (let tag of tags) {
 		let data = tagColors.get(tag.tag);
-		if (data) {
+		let containsEmoji = Zotero.Utilities.Internal.containsEmoji(tag.tag);
+		// Include tags that contain emojis even if they are not assigned a color
+		if (data || containsEmoji) {
```

Comment:
Doesn't seem right that this is still called `getColoredTags()`. Since it will now return both colored tags and emoji tags, we should probably rename it to `getItemsListTags()`.

And then we're now sorting against `undefined` below for emoji tags, which means the comparison function is actually returning NaN, which is equivalent to a return value of `0` in JS, which means that emoji tags without a position defined can be sorted in the middle of the list. We need to check for `.position` and return tags with positions first. Tags without positions should return 0 against each other to remain in their sorted order. (And we should add a test for this sorting.)

Reply:
Right! I added proper sorting in the last commit. I thought it may be more readable to aggregate and sort all colored tags (just like it is done now) first and then just `concat` tags with emojis to the end of that array instead of collecting all tags into one array and sorting them together.